lib/commands/cryptographic-identities.ts

```diff
@@ -602,11 +604,32 @@ export class ImportCryptographicIdentity implements ICommand {
 			}
 
 			if(!password && importType === CryptographicIdentityConstants.PKCS12CERTIFICATE) {
-				password = this.$prompter.getPassword("Certificate file password", { allowEmpty: true }).wait();
+				isPasswordRequired = true;
 			}
 
-			var targetFile = this.$fs.createReadStream(certificateFile);
-			var result = this.$server.identityStore.importIdentity(<any>importType, password, targetFile).wait();
+			var targetFile : any;
+			var result : Server.CryptographicIdentityData[];
+
+			for (var i = 0; i < CryptographicIdentityConstants.MAX_ALLOWED_PASSWORD_ATTEMPTS; ++i) {
+				var noErrorOccurred = true;
+				
+				targetFile = this.$fs.createReadStream(certificateFile);
+				if (isPasswordRequired) {
+					password = this.$prompter.getPassword("Certificate file password", { allowEmpty: true }).wait();
+				}
+
+				try {
+					result = this.$server.identityStore.importIdentity(<any>importType, password, targetFile).wait();
+				}
+				catch(err) {
+					noErrorOccurred = false;
```

Comment:
what will happen if the problem with the import is not from the password? will you still ask for password 3 times?